### PR TITLE
feat: introduce plans API

### PR DESCRIPTION
**Pull request by kjaskiewiczz**
Ticket: [ALV-116](https://northerntech.atlassian.net/browse/ALV-116)

[ALV-116]: https://northerntech.atlassian.net/browse/ALV-116?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by kjaskiewiczz** [comment]
@mender-test-bot start pipeline --fast

**Comment by kjaskiewiczz** [comment]
@mender-test-bot start pipeline --fast

**Comment by kjaskiewiczz** [comment]
@mender-test-bot start pipeline --fast

File: model/plans.go [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+	Product string `json:"product" bson:"product"`
+
+	// name to display
+	DisplayName string `json:"display_name" bson:"display_name"`
+
+	// feature set
+	Features *Features `json:"features" bson:"features"`
+
+	// limits
+	Limits *PlanLimits `json:"limits,omitempty" bson:"limits,omitempty"`
```
Comment (alfrunes) [inline model/plans.go:83]: As discussed, maybe we should keep the Limits only in the "PlanBinding" object.
```suggestion
```

File: docs/management_api.yml [hunk 2]
```diff
@@ -698,3 +768,102 @@ definitions:
...
+      display_name:
+        description: |
+            Short information about the plan.
+        type: string
+      features:
+        $ref: "#/definitions/Features"
+      limits:
+        $ref: "#/definitions/PlanLimits"
+    required:
+      - product
```
Comment (alfrunes) [inline docs/management_api.yml:792]: Is `product` really necessary?

Comment (kjaskiewiczz) [reply docs/management_api.yml]: I don't think so; removed

File: model/plans.go (hunk 1 above)
Comment (kjaskiewiczz) [reply model/plans.go:80]: done

File: docs/management_api.yml [hunk 3]
```diff
@@ -551,6 +551,76 @@ paths:
...
+        - Management API
+      security:
+        - ManagementJWT: []
+      summary: Get plan and limits information for current tenant
+      parameters:
+        - name: id
+          in: path
+          type: string
+          description: Object identifier.
+          required: true
```
[the remark this answers is not in the document]
Comment (kjaskiewiczz) [reply docs/management_api.yml]: leftover from previous proposal; fixed;
thanks

File: model/plans.go (hunk 1 above)
Comment (alfrunes) [reply model/plans.go:80]: Sorry, what did you do about the limits? Maybe I'm missing something :thinking:

**Review by alfrunes — COMMENTED** [review]
Your commit message needs some fixup, and I'm still seeing the limits in the plan object. Is that intentional?

File: model/plans.go (hunk 1 above)
Comment (kjaskiewiczz) [reply model/plans.go]: right; I forgot to include the last change and rebase... sorry; fixed now

**Review by alfrunes — APPROVED** [review]

**Comment by kjaskiewiczz** [comment]
@mender-test-bot start pipeline --fast

**Pull request merged**